pkg/detector/compare.go

```diff
@@ -9,35 +9,70 @@ import (
 	"github.com/karmada-io/karmada/pkg/util/fedinformer/keys"
 )
 
-func getHighestPriorityPropagationPolicies(policies []*policyv1alpha1.PropagationPolicy, resource *unstructured.Unstructured, objectKey keys.ClusterWideKey) *policyv1alpha1.PropagationPolicy {
+func getHighestPriorityPropagationPolicy(policies []*policyv1alpha1.PropagationPolicy, resource *unstructured.Unstructured, objectKey keys.ClusterWideKey) *policyv1alpha1.PropagationPolicy {
+	matchedPolicyImplicitPriority := util.PriorityMisMatch
+	var matchedPolicyExplicitPriority int32
 	var matchedPolicy *policyv1alpha1.PropagationPolicy
-	matchedPolicyPriority := util.PriorityMisMatch
+
 	for _, policy := range policies {
-		if p := util.ResourceMatchSelectorsPriority(resource, policy.Spec.ResourceSelectors...); p > matchedPolicyPriority {
+		implicitPriority := util.ResourceMatchSelectorsPriority(resource, policy.Spec.ResourceSelectors...)
+		if implicitPriority <= util.PriorityMisMatch {
+			continue
+		}
+		explicitPriority := policy.ExplicitPriority()
+
+		if matchedPolicy == nil || matchedPolicyExplicitPriority < explicitPriority {
```

Comment:
Since `matchedPolicyExplicitPriority` defaults to 0, but the `policy.ExplicitPriority()` could be < 0, in that case, the policy will be missed.

By the way, `matchedPolicy == nil` seems redundant.

Reply:
Emm, you are right.  My assumption is based on removing the `matchedPolicy == nil`.

I think we can default `matchedPolicyExplicitPriority` to `math.MinInt32` and remove `matchedPolicy == nil` would be more clear.